presto-cache/src/main/java/com/facebook/presto/cache/CacheManager.java

```diff
@@ -22,13 +24,16 @@
 {
     /**
      * Given {@param request}, check if the data is in cache.
-     * If it is not in cache, return false.
-     * Otherwise, save the data in {@param buffer} starting at {@param offset} and return true.
+     * If it is in cache, save the data in {@param buffer} starting at {@param offset} and return IN_CACHE.
+     * If it is not in cache:
+     *      1. If there is still cache quota for this table, return NOT_IN_CACHE
+     *      2. Otherwise, return CACHE_QUOTA_EXCEED
+     * @return CacheResult
      */
-    boolean get(FileReadRequest request, byte[] buffer, int offset);
+    CacheResult get(CacheQuota cacheQuota, FileReadRequest request, byte[] buffer, int offset);
 
     /**
      * Save data in cache
      */
-    void put(FileReadRequest request, Slice data);
+    void put(long cacheScopeIdentifier, FileReadRequest request, Slice data);
```

Comment:
- We should use `CacheQuota` as well to match the one in `get`
- Move `cacheQuota` to the last param